### ISSUE TITLE
What does lrf do when cos_lr = True?

### ISSUE THREAD

**Issue by Lakror** [ultralytics/ultralytics#6256]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello @glenn-jocher or others,

First of all: loving YOLOV8, great work!

I am tuning my hyperparameters but I came across contradictory information concerning the role of lrf when cos_lr is activated.

According to: https://github.com/ultralytics/ultralytics/issues/2849#issuecomment-1565227719 the lrf is still of importance in this case, while https://github.com/ultralytics/ultralytics/issues/2026#issuecomment-1508592203 says that cos_lr will be used instead of lrf when activated.

Could someone clarify?

**Comment by glenn-jocher** [comment]
@Lakror hello,

Thank you for your kind words! Regarding your question on the `lrf` (learning rate final) parameter when `cos_lr` is set to `True`, it's important to understand how the cosine learning rate scheduler functions. The `cos_lr` sets a cosine annealing schedule for the learning rate, which means the learning rate will follow a cosine curve between an initial learning rate and a final learning rate over the course of training.

When `cos_lr` is enabled, the `lrf` parameter still comes into play as it designates the final learning rate value that the cosine annealing scheduler will target at the end of the training process. So, even with `cos_lr` set to `True`, the value of `lrf` is important as it determines the lower boundary of the learning rate towards which the scheduler will anneal.

I hope this provides clarity on how the `lrf` is utilized in conjunction with the cosine learning rate scheduler in YOLOv8. For more detailed explanations on hyperparameter settings, please refer to our documentation at the Ultralytics Docs.
